xnmt/reports.py

```diff
@@ -62,16 +76,30 @@ def add_sent_for_report(self, sent_info: Dict[str,Any]) -> None:
       self._sent_info_list = []
     self._sent_info_list.append(sent_info)
 
+  def report_global_info(self, glob_info: Dict[str, Any]) -> None:
```

Comment:
I'm a little bit hesitant about "Globals", as I've found that these can break modularity and add complex data dependencies. That being said, this might be the best solution here? If you don't think there's another better way, we can go with this.

Reply:
Yes I agree it's not a very good design, but this is a problem with the whole reporting mechanism which spreads global information on either a sentence base (```report_sent_info()```) or corpus base (```report_global_info()```). Should I just rename the latter to ```report_corpus_info```, and leave improvements to the reporting to a future PR? I honestly don't have a good idea right now for how to make it more modular without adding complexity to it.

Reply:
Sure.